### PR TITLE
New binding base object

### PR DESCRIPTION
**Pull request by Wall-E-76**
All the functions binded in SofaPython for BaseObject are now also binded in SofaPython3.
Every function has docstring documentation, and the bindings are tested.
A new component was created : Binding_BaseObject_MockComponent, that inherits from BaseObject. It is used to test the bindings of virtual functions of BaseObject. It has only one attribute, a data string. In order to test a virtual function, the function is implemented in the Mock Component, and simply changes the value of the data string to the name of the function itself. That way, it is simple to check in the test script in Python whether the function was called or not, by simply looking at the value of the data string.

### CODE REVIEW

**Comment by marques-bruno** [comment]
Thanks for your contribution!
Could one of the admins verify this patch?

File: bindings/Sofa/src/SofaPython3/Sofa/Core/Binding_BaseObject.cpp [hunk 1]
```diff
@@ -19,6 +46,10 @@ py::object getItem(const BaseObject& self, const std::string& path)
     throw py::value_error("Invalid syntax"); // should never get there
 }
 
+std::string getLinkPath(const BaseObject *self, py::arg args){
+    return std::string("@")+self->getPathName().c_str();
```
Comment (damienmarchal) [inline bindings/Sofa/src/SofaPython3/Sofa/Core/Binding_BaseObject.cpp]: I think the c_str() is not needed.

**Comment by damienmarchal** [comment]
Hi,

Congratulation for this PR as it is non trivial and well done. 
I'm going to merge it when the minor comment is fixed

**Comment by marques-bruno** [comment]
[ci-build]

**Comment by marques-bruno** [comment]
[ci-build]

**Comment by damienmarchal** [comment]
[ci-build]

**Pull request merged**